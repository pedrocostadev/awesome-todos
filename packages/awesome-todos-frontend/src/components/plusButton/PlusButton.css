.plus-button {
	display: flex;
	align-items: center;
	margin-right: var(--spacing-xs);
	justify-content: center;
	background-color: var(--green);
	border: none;
}

.plus-button:hover {
	cursor: pointer;
}
