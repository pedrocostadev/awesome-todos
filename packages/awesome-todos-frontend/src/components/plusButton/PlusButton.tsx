import React from 'react';
import PlusIcon from './PlusIcon';
import './PlusButton.css';

interface Props {
  onClick(): void;
}

const PlusButton: React.FC<Props> = ({ onClick }) => {
  return (
    <button onClick={onClick} className="plus-button" data-testid="add-todo-button">
      <PlusIcon />
    </button>
  );
};

export default PlusButton;
