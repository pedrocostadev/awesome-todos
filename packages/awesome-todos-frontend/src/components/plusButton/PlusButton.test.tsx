import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import PlusButton from './PlusButton';

describe('<PlusButton />', () => {
  const onClick = jest.fn();
  test('It should render', () => {
    const button = render(<PlusButton onClick={onClick} />);
    expect(button).toMatchSnapshot();
  });
  test('It should call onClick', () => {
    const { getByTestId } = render(<PlusButton onClick={onClick} />);
    const button = getByTestId('add-todo-button');
    fireEvent.click(button);
    expect(onClick).toBeCalledTimes(1);
  });
});
