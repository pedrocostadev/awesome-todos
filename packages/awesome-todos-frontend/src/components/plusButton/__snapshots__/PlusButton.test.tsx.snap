// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlusButton /> It should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="plus-button"
        data-testid="add-todo-button"
      >
        <svg
          fill="white"
          height="24"
          stroke="white"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            d="M24 10h-10v-10h-4v10h-10v4h10v10h4v-10h10z"
          />
        </svg>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="plus-button"
      data-testid="add-todo-button"
    >
      <svg
        fill="white"
        height="24"
        stroke="white"
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M24 10h-10v-10h-4v10h-10v4h10v10h4v-10h10z"
        />
      </svg>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
